spec/requests/api/v1/carts/remove_product_spec.rb

```diff
@@ -0,0 +1,64 @@
+describe 'PUT /api/v1/carts/remove_product', type: :request do
+  subject { put remove_product_api_v1_carts_path, headers: headers, params: params, as: :json }
+
+  let(:headers) { auth_headers }
+  let(:user) { create(:user) }
+  let(:params) do
+    {
+      cart: {
+        product_id: product.id,
+        quantity: quantity
+      }
+    }
+  end
+
+  context 'when response is successful' do
+    let!(:cart_product) do
+      create(:cart_product, cart: cart, product: product, quantity: 2, total_amount: total_amount)
+    end
+    let(:cart) { create(:cart, user: user, total_items: 2, total_price: 20) }
+    let(:product) { create(:product, price: 10) }
```

Comment:
in order to improve the readability, i'd move the `cart` and `product` definitions above the `cart_product` one 
```suggestion
    let(:cart) { create(:cart, user: user, total_items: 2, total_price: 20) }
    let(:product) { create(:product, price: 10) }
    let!(:cart_product) do
      create(:cart_product, cart: cart, product: product, quantity: 2, total_amount: total_amount)
    end
```

Reply:
Updated